lib/hyrax/acl/access_control.rb

```diff
@@ -36,8 +36,8 @@ class AccessControl < Valkyrie::Resource
       #
       # @return [AccessControl]
       # @raise [ArgumentError] if the resource is not persisted
-      def self.for(resource:, query_service: Hyrax.query_service)
-        query_service.custom_queries.find_access_control_for(resource: resource)
+      def self.for(resource:, query_service:)
+        Hyrax::Acl::CustomQueries::FindAccessControl.new(query_service: query_service).find_access_control_for(resource: resource)
       rescue Valkyrie::Persistence::ObjectNotFoundError
         new(access_to: resource.id)
```

Comment:
Do we want to add empty `permissions: []` to avoid the nil error when checking permissions? Here is codes that we are using with Superskunk integration that seems to be working: https://gitlab.com/surfliner/surfliner/-/commit/2f59dab50f15218fab9e6a60ab0df1635cae7ec2#d991336890301b37c30491fd44d83097a571aabf_26_21.